file/replica_client.go

```
@@ -117,13 +121,29 @@ func (c *ReplicaClient) OpenLTXFile(ctx context.Context, level int, minTXID, max
 	return f, nil
 }
 
-// WriteLTXFile writes an LTX file to disk.
+// WriteLTXFile writes an LTX file to the replica.
+// Extracts timestamp from LTX header and sets it as the file's ModTime to preserve original creation time.
 func (c *ReplicaClient) WriteLTXFile(ctx context.Context, level int, minTXID, maxTXID ltx.TXID, rd io.Reader) (info *ltx.FileInfo, err error) {
 	var fileInfo, dirInfo os.FileInfo
 	if db := c.db(); db != nil {
 		fileInfo, dirInfo = db.FileInfo(), db.DirInfo()
 	}
 
+	// Use TeeReader to peek at LTX header while preserving data for upload
+	var buf bytes.Buffer
+	teeReader := io.TeeReader(rd, &buf)
+
+	// Extract timestamp from LTX header
+	var timestamp time.Time
+	if hdr, _, err := ltx.PeekHeader(teeReader); err == nil {
+		timestamp = time.UnixMilli(hdr.Timestamp).UTC()
+	} else {
+		timestamp = time.Now().UTC() // Fallback if header read fails
+	}
```

Return error instead of fallback.